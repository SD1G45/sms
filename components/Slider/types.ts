import React from "react";

export interface SliderProps {
  value: number;
  onChange: (event: React.ChangeEvent<HTMLInputElement>) => void;
}
