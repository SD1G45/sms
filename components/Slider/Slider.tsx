import React from "react";
import styled from "styled-components";
import { SliderProps } from "./types";

const sliderThumbStyles = (props: any) => `
  width: 25px;
  height: 25px;
  background: ${props.theme.colors.primary};
  border-radius: 50%;
  cursor: pointer;
  -webkit-transition: .2s;
  transition: opacity .2s;
`;

const Styles = styled.div`
  display: flex;
  align-items: center;
  color: #888;
  margin-top: 2rem;
  margin-bottom: 2rem;
  .value {
    flex: 1;
    font-size: 2rem;
  }
  .slider {
    flex: 6;
    -webkit-appearance: none;
    width: 100%;
    height: 15px;
    border-radius: 5px;
    background: #efefef;
    outline: none;
    &::-webkit-slider-thumb {
      -webkit-appearance: none;
      appearance: none;
      ${(props) => sliderThumbStyles(props)}
    }
    &::-moz-range-thumb {
      ${(props) => sliderThumbStyles(props)}
    }
  }
`;

const Slider: React.FC<SliderProps> = ({ value, onChange }) => {
  return (
    <Styles>
      <input
        type="range"
        min={0}
        max={100}
        value={value}
        className="slider"
        onChange={onChange}
      />
    </Styles>
  );
};

export default Slider;
