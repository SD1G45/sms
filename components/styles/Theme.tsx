import React from "react";
import { ThemeProvider } from "styled-components";
import ITheme from "./types";

const screenWidth = {
  minSmall: `(min-width: ${430}px)`,
  minMedium: `(min-width: ${730}px)`,
  minLarge: `(min-width: ${1024}px)`,
  maxSmall: `(max-width: ${430}px)`,
  maxMedium: `(max-width: ${730}px)`,
  maxLarge: `(max-width: ${1024}px)`,
};

const lightTheme: ITheme = {
  colors: {
    primary: "#4881F0",
    lightGray: "#BEC8D2",
    textPrimary: "#000000",
    error: "#FF7F7F",
  },
  screenWidth: { ...screenWidth },
};

interface ThemeProps {
  children: React.ReactNode;
}

const Theme: React.FC<ThemeProps> = ({ children }) => (
  <ThemeProvider theme={lightTheme}>{children}</ThemeProvider>
);

export default Theme;
