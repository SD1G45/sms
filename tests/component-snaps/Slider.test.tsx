import React from "react";
import renderer from "react-test-renderer";
import "jest-styled-components";
import Slider from "../../components/Slider";
import Theme from "../../components/styles/Theme";

describe("<Slider/>", () => {
  it("renders correctly", () => {
    const tree = renderer.create(
      <Theme>
        <Slider value={1} onChange={() => {}}/>
      </Theme>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});