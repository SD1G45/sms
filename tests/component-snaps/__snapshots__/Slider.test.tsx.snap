// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Slider/> renders correctly 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  color: #888;
  margin-top: 2rem;
  margin-bottom: 2rem;
  width: 100%;
}

.c0 .value {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  font-size: 2rem;
}

.c0 .slider {
  -webkit-flex: 6;
  -ms-flex: 6;
  flex: 6;
  -webkit-appearance: none;
  width: 100%;
  height: 15px;
  border-radius: 5px;
  background: #efefef;
  outline: none;
}

.c0 .slider::-webkit-slider-thumb {
  -webkit-appearance: none;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  width: 25px;
  height: 25px;
  background: #4881F0;
  border-radius: 50%;
  cursor: pointer;
  -webkit-transition: .2s;
  -webkit-transition: opacity .2s;
  transition: opacity .2s;
}

.c0 .slider::-moz-range-thumb {
  width: 25px;
  height: 25px;
  background: #4881F0;
  border-radius: 50%;
  cursor: pointer;
  -webkit-transition: .2s;
  -webkit-transition: opacity .2s;
  transition: opacity .2s;
}

<div
  className="c0"
>
  <input
    aria-labelledby="continuous-slider"
    className="slider"
    max={5}
    min={1}
    onChange={[Function]}
    step={0.001}
    type="range"
    value={1}
  />
</div>
`;
